iotedgedev/deploymentmanifest.py

```diff
@@ -147,3 +186,74 @@ def _get_module_content(self):
+            deployment_schema = json.loads(urlopen(Constants.deployment_manifest_schema_url).read())
+            validator_class = jsonschema.validators.validator_for(deployment_schema)
+            validator = validator_class(deployment_schema)
+            validation_errors = validator.iter_errors(self.json)
+            error_detected = False
+            for error in validation_errors:
+                error_detected = True
+                self.output.info("Deployment manifest schema error: %s. Property path:%s" % (error.message, "->".join(error.path)))
+            if error_detected:
+                self.output.info("Deployment manifest schema validation failed. Please see previous logs for more details")
+            else:
+                self.output.info("Deployment manifest schema validation passed.")
+        except jsonschema.exceptions.SchemaError as schemaErr:
+            self.output.info("Errors found in deployment manifest schema, skip schema validation. Error:%s" % schemaErr.message)
+        except Exception as ex:  # Ignore other non schema validation errors
+            self.output.info("Unexpected error during deployment manifest schema validation, skip schema validation. Error:%s" % ex)
+
+        return validation_success
+
+    # Call _validate_deployment_manifest_schema first. This function assumes createOptions are strings.
+    def _validate_createOptions(self):
+        self.output.info("Start validating createOptions for all modules.")
+        modules = self.get_all_modules()
+        validation_success = True
+        for module_name, module_info in modules.items():
+            try:
+                self.output.info("Validating createOptions for module %s" % module_name)
+                create_options = []
+                if "settings" in module_info and "createOptions" in module_info["settings"]:
+                    create_options_value = module_info["settings"]["createOptions"]
+                    if len(create_options_value) > TWIN_VALUE_MAX_SIZE:
+                        validation_success = False
+                        self.output.info("Length of createOptions in module %s exceeds %d" % (module_name, TWIN_VALUE_MAX_SIZE))
+                    create_options.append(create_options_value)
+                    # Merge additional create options
+                    for i in range(1, TWIN_VALUE_MAX_CHUNKS):
+                        property_name = "createOptions0%d" % i
+                        if property_name in module_info["settings"]:
+                            create_options_value = module_info["settings"][property_name]
+                            if len(create_options_value) > TWIN_VALUE_MAX_SIZE:
+                                validation_success = False
+                                self.output.info("Length of %s in module %s exceeds %d" % (property_name, module_name, TWIN_VALUE_MAX_SIZE))
+                            create_options.append(create_options_value)
```

Comment:
if exceed still continue top append?

Reply:
Yes, so we can check whether the string is still a valid json. Though the user need to split the createOptions later.
This is for following scenario: If the createOptions exceeds max length and also is not a valid json, we could report 2 errors to user instead of telling user max length exceeded error first and then tell user not valid json error in the second run.